mqt/qecc/bindings.cpp

```diff
@@ -24,7 +35,77 @@ std::vector<bool> sampleIidPauliErr(const std::size_t length, const double physi
     return Utils::sampleErrorIidPauliNoise(length, physicalErrRate);
 }
 
+py::dict applyEcc(const py::object& circ, const std::string& eccName, const size_t eccFrequency) {
+    auto qc = std::make_shared<qc::QuantumComputation>();
+
+    try {
+        if (py::isinstance<py::str>(circ)) {
+            auto&& file = circ.cast<std::string>();
+            qc->import(file);
+        } else {
+            py::object const quantumCircuit       = py::module::import("qiskit").attr("QuantumCircuit");
+            py::object const pyQasmQobjExperiment = py::module::import("qiskit.qobj").attr("QasmQobjExperiment");
+            if (py::isinstance(circ, quantumCircuit)) {
+                qc::qiskit::QuantumCircuit::import(*qc, circ);
+            } else if (py::isinstance(circ, pyQasmQobjExperiment)) {
+                qc::qiskit::QasmQobjExperiment::import(*qc, circ);
+            }
+        }
+    } catch (std::exception const& e) {
+        std::stringstream ss{};
+        ss << "Could not import circuit: " << e.what();
+        return py::dict("error"_a = ss.str());
+    }
```

Comment:
Remove this try catch block. Exceptions are automatically translated to python exceptions when they are encountered.
Alternatively, rethrow the exception instead of returning an error.

Reply:
I removed the try catch block